lib/Codegen/Lowering/IdentifierError.cpp

```diff
@@ -0,0 +1,110 @@
+#include "marco/Codegen/Lowering/IdentifierError.h"
+#include "marco/Codegen/Lowering/Distance/EditDistance.h"
+#include "marco/Codegen/Lowering/Distance/SentenceDistanceCalculator.h"
+
+namespace marco::codegen::lowering
+{
+  IdentifierError::IdentifierError(const IdentifierType &identifierType, std::string identifierName, 
+        const std::set<std::string> &declaredIdentifiers, unsigned int line, unsigned int column): 
+        identifierType(identifierType), actualName(identifierName), predictedName(""),
+        line(line), column(column) {
+    unsigned int lowestDistance = UINT32_MAX;
+
+    // Choose which built in identifiers to check.
+    const std::set<std::string> emptySet = {};
```

Comment:
This is an ugly hack. Call appropriate functions passing the needed set or iterators.